### PR TITLE
Split namespace packages and add httpx client

### PR DESCRIPTION
**Pull request by n4mespace**
Changelog:

- updated project deps
- add ruff
- move grpc part to `featureflags/grpc/`
- add http part in `featureflags/http/` 
- some code duplicates, like in `conditions.py` because a lot was written with types from protobufs
- add httpx manager `featureflags/http/managers/httpx.py` and example for it `examples/http/httpx_client.py`

QA:

- from which version we plan to support clients?
if not from `≥3.7` then we can use more modern typing (just run `pyupgrade`)

TODO:

- add sync `requests`/ async `aiohttp` managers
- update protobuf version in deps after protobuf release
- update tests/enable test workflow
- regenerate docs
- add pre-commit hooks
- try to integrate with some service http API
- add `tracer` / `stats_collector` to http managers

### CODE REVIEW

File: featureflags/grpc/managers/asyncio.py [hunk 1]
```diff
@@ -1,13 +1,25 @@
...
-from ..state import State
-
+from asyncio import AbstractEventLoop
+from typing import Callable, Dict, List, Optional
+
+from featureflags.grpc.managers.base import AbstractManager
+from featureflags.grpc.state import GrpcState
+from featureflags.grpc.stats_collector import StatsCollector
+from featureflags.grpc.tracer import Tracer
+from featureflags.grpc.types import Variable
```
Comment (kindermax) [inline featureflags/grpc/managers/asyncio.py:11]: I think from `featureflags.grpc.types` and `from featureflags.http.types` must not be duplicated. This must live somewhere in `from featureflags_client.types`

File: examples/grpc/aiohttp_app.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
+import logging
+
+import flags
+from aiohttp import web
+from grpclib.client import Channel
+
+from featureflags.grpc.client import FeatureFlagsClient
```
Comment (kindermax) [inline examples/grpc/aiohttp_app.py:7]: Should we rename `featureflags` to `featureflags_client` ?

File: featureflags/http/managers/httpx.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        return self._state.get(name)
+
+    def start(self) -> None:
+        if self._refresh_task is not None:
+            raise RuntimeError("Manager is already started")
+
+        self._refresh_task = asyncio.create_task(self._refresh_loop())
+
+    async def wait_closed(self) -> None:
+        self._refresh_task.cancel()
```
Comment (kindermax) [inline featureflags/http/managers/httpx.py:117]: In grpc manager, there is a separate method `close` which closes refresh_task. Lets try following that implementation. (Or if all in one method is desired we can rename it to `async close`)

File: .github/workflows/release.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+jobs:
+  deploy-client:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./client
+    if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags/client-')
+    strategy:
+      matrix:
+        python-version: [3.7]
```
Comment (n4mespace) [inline .github/workflows/release.yaml:21]: by the way, @kindermax, what minimun version we should support? can we start from 3.9/3.10 at least? maybe do a poll in dev channel?

**Pull request merged**